roofit/hs3/src/JSONFactories_HistFactory.cxx

```diff
@@ -366,37 +376,33 @@ bool importHistSample(RooJSONFactoryWSTool &tool, RooDataHist &dh, Scope &scope,
                                                            : "alpha_" + sysname);
             RooAbsReal *par = ::getNP(ws, parname);
             histo_nps.add(*par);
-            RooArgSet varlist;
-            scope.getObservables(varlist);
-            auto dh_low = RooJSONFactoryWSTool::readBinnedData(p["data"], sysname + "Low_" + nameWithPrefix, varlist);
-            ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "Low_" + nameWithPrefix, *dh_low->get(), *dh_low));
-            histo_low.add(*ownedArgsStack.top());
-            auto dh_high = RooJSONFactoryWSTool::readBinnedData(p["data"], sysname + "High_" + nameWithPrefix, varlist);
-            ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "High_" + nameWithPrefix, *dh_high->get(), *dh_high));
-            histo_high.add(*ownedArgsStack.top());
+	    auto &data = mod["data"];
+            ownedHistStack.push(std::move(RooJSONFactoryWSTool::readBinnedData(data["lo"], sysname + "Low_" + nameWithPrefix, varlist)));
+	    ownedArgsStack.push(makeUnique<RooHistFunc>(sysname + "Low_" + nameWithPrefix, varlist, *ownedHistStack.top()));
+	    histo_low.add(*ownedArgsStack.top());
+	    ownedHistStack.push(std::move(RooJSONFactoryWSTool::readBinnedData(data["hi"], sysname + "High_" + nameWithPrefix, varlist)));				
```

Comment:
FYI, it's not meanigful to do `std::move` if it's a temporary anyway. Because without it, the compiler will already figure out what is the best thing to do, and in fact using `std::move` in these cases results in a warning as we saw in the CI.

Reply:
fixed.